plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotIntegrationConnectorSmokeTest.java

```diff
@@ -2439,8 +2428,23 @@ public void testVarbinary()
     @Test
     public void testTimestamp()
     {
-        assertThat(query("SELECT ts FROM alltypes order by ts LIMIT 1")).matches("VALUES (TIMESTAMP '1970-01-01 00:00:00.000')");
-        assertThat(query("SELECT min(ts) FROM alltypes")).matches("VALUES (TIMESTAMP '1970-01-01 00:00:00.000')");
+        assertThat(query("SELECT ts FROM " + ALL_TYPES_TABLE + " ORDER BY ts LIMIT 1")).matches("VALUES (TIMESTAMP '1970-01-01 00:00:00.000')");
+        assertThat(query("SELECT min(ts) FROM " + ALL_TYPES_TABLE)).matches("VALUES (TIMESTAMP '1970-01-01 00:00:00.000')");
+        assertThat(query("SELECT ts FROM " + ALL_TYPES_TABLE + " ORDER BY ts DESC LIMIT 1")).matches("SELECT max(ts) FROM " + ALL_TYPES_TABLE);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.of("UTC"));
+        for (int i = 0, step = 1200; i < MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES - 2; i++) {
+            String initialUpdatedAtStr = formatter.format(initialUpdatedAt.plusMillis(i * step));
+            assertThat(query("SELECT ts FROM " + ALL_TYPES_TABLE + " WHERE ts >= TIMESTAMP '" + initialUpdatedAtStr + "' ORDER BY ts LIMIT 1"))
+                    .matches("VALUES (TIMESTAMP '" + initialUpdatedAtStr + "')");
+            assertThat(query("SELECT ts FROM " + ALL_TYPES_TABLE + " WHERE ts <= TIMESTAMP '" + initialUpdatedAtStr + "' ORDER BY ts DESC LIMIT 1"))
+                    .matches("VALUES (TIMESTAMP '" + initialUpdatedAtStr + "')");
+            assertThat(query("SELECT ts FROM " + ALL_TYPES_TABLE + " WHERE ts = TIMESTAMP '" + initialUpdatedAtStr + "' LIMIT 1"))
+                    .matches("VALUES (TIMESTAMP '" + initialUpdatedAtStr + "')");
+        }
+        assertThat(query("SELECT timestamp_col FROM " + ALL_TYPES_TABLE + " WHERE timestamp_col < TIMESTAMP '1971-01-01 00:00:00.000'")).isFullyPushedDown();
+        assertThat(query("SELECT timestamp_col FROM " + ALL_TYPES_TABLE + " WHERE timestamp_col < TIMESTAMP '1971-01-01 00:00:00.000'")).matches("VALUES (TIMESTAMP '1970-01-01 00:00:00.000'), (TIMESTAMP '1970-01-01 00:00:00.000')");
```

Comment:
IIRC `isFullyPushedDown` verifies correctness by default by running query twice - once with pushdown enabled and once disabled. (unless you disable this using `skipResultsCorrectnessCheckForPushdown()`).

Reply:
hmm, so I don't need to compare the response, just use `isFullyPushedDown` for all the query?

Reply:
Yes.

Reply:
updated.